js/modules/k6/execution/execution.go

```diff
@@ -215,14 +215,19 @@ func (mi *ModuleInstance) newVUInfo() (*goja.Object, error) {
 	tags := &tagsDynamicObject{runtime: rt, state: vuState}
 	tagsObj := rt.NewDynamicObject(tags)
 
-	// TODO: change the prototype of the tags dynamic object so it has a new
+	// We change the prototype of the tags dynamic object, so it has a new
 	// isIndexed(key) method that returns true if the specific tag is indexed
-	// (stored in the TagSet) or false if not (stored in the metadata string
-	// map) or undefined if the tag doesn't exist?
-	// tagsObj.SetPrototype(...)
+	// (stored in the TagSet), or false if not (stored in the metadata string
+	// map), or undefined if the tag doesn't exist.
+	proto := rt.NewObject()
+	if err := proto.Set("isIndexed", tags.isIndexed); err != nil {
+		return o, err
+	}
+	if err := tagsObj.SetPrototype(proto); err != nil {
```

Comment:
Is there a particular reason to be on the Prototype ? Not that it's wrong I am just asking if you shouldn't just do `o.Set` instead :shrug:

Reply:
A couple of reasons. If I try, because of the `goja.DynamicObject` validation logic for `Set()`, I will get this error :sweat_smile: 
```
TypeError: setting 'isIndexed' metric tag failed; only String, Boolean and Number types, or objects with the 'value' and 'index' properties, are accepted as a metric tag values
```

But even if I work around this, I think it's better if `Object.keys(execution.vu.tags)` doesn't return `isIndexed`

Reply:
Hmm I would argue this needs to go in `tagsDynamicObject#Get` and you should check `isIndexed` and return  ToValue(o.isIndexed)` isntead of adding a whole new prototyupe

Reply:
Hm OK, I can probably do that, but it feels like a hack and I don't understand what the problem with the prototype is? :confused:

Reply:
> it feels like a hack 

Well the whole DynamicObject is built on this hack ;)

> I don't understand what the problem with the prototype is? 

You can definitely do it, but:
1. you don't need to as there is an easier way
2. you should  use [DefinedAccessorProperty](https://pkg.go.dev/github.com/dop251/goja#Object.DefineAccessorProperty) and set `enumerable` to goja.FLAG_False as otherwise `for (let l in exec.vu.tags)` will still catch it (even though it is won't come in `Object.keys`)
3. I would prefer that when we start setting prototypes we do actually create *one* prototype object and set it across althe instances of that particular "type". In this case this is likely an overkill, but so is using a prototype ;)

Reply:
hmm I tried to do this, but now I get this error :sweat_smile: 
```
TypeError: Dynamic objects do not support accessor properties
```

Reply:
I meant `2` as in if you are using prototype - you should use `Define...` instead of `Set`

Reply:
Ah, I see... :man_facepalming: :+1: I will try to make a single prototype per VU/runtime and use that then.

Reply:
Resolved by [this force-pushed fixup](https://github.com/grafana/k6/compare/c8bfdd7f722411ed6846fe41d626ab07f618712f..7c90538db8b799df11022500cb81d91c4e223dc1) to the second commit. 

I don't particularly like how now I have to check if the `state` is `nil` everywhere, but overall I agree it's better to have a single prototype (and even `tags` object) per VU :bow: Besides, I think there aren't any actual technical obstacles for the `exec.vu.tags` (and other `exec.vu` properties) to work even in the init context, with some minor refactoring in the future... :thinking: I think we don't currently do it, because the `Tags`, `VUID` and `VUIDGlobal` properties of the VU are being kept in the `lib.State` struct, but that doesn't really need to be the case :thinking: These can be made available even in the init context, I think, so we may choose to relax this constraint in the future.

Reply:
I meant just the prototype ... but  I also was under the wrong impression that the `tags` were created once ... so I guess this is an improvement as well, so I will resolve it.